aeon/transformations/panel/dilated_shapelet_transform.py

```
@@ -0,0 +1,895 @@
+                ]
+
+            if len(self.shapelet_lengths_) == 0:
+                raise ValueError(
+                    "Shapelet lengths array is empty, did you give shapelets lengths"
+                    " superior to the size of the series ?"
+                )
+
+        self.threshold_percentiles_ = self.threshold_percentiles
+        if self.threshold_percentiles_ is None:
+            self.threshold_percentiles_ = np.array([5, 10])
+        else:
+            if not isinstance(self.threshold_percentiles_, (list, tuple, np.ndarray)):
+                raise TypeError(
+                    "Expected a list, numpy array or tuple for threshold_percentiles"
+                )
+            if len(self.threshold_percentiles_) != 2:
+                raise ValueError(
+                    "The threshold_percentiles param should be an array of size 2"
+                )
+            self.threshold_percentiles_ = np.asarray(self.threshold_percentiles_)
+
+    @classmethod
+    def get_test_params(cls, parameter_set="default"):
+        """Return testing parameter settings for the estimator.
+
+        Parameters
+        ----------
+        parameter_set : str, default="default"
+            Name of the set of test parameters to return, for use in tests. If no
+            special parameters are defined for a value, will return `"default"` set.
+            There are currently no reserved values for transformers.
+
+        Returns
+        -------
+        params : dict or list of dict, default = {}
+            Parameters to create testing instances of the class
+            Each dict are parameters to construct an "interesting" test instance, i.e.,
+            `MyClass(**params)` or `MyClass(**params[i])` creates a valid test instance.
+            `create_test_instance` uses the first (or only) dictionary in `params`
+        """
+        if parameter_set == "default":
+            params = {"max_shapelets": 10}
```

Not a change request, but an alternative to using the minimum series length as the shapelet length if its too small is to just set a very small shapelet length here for testing.

---

This is only if you want to throw an error for invalid length inputs rather than adjust to smaller series.